### ISSUE TITLE
Extension causes high cpu load

### ISSUE THREAD

**Issue by SterlingKinder** [juanfranblanco/vscode-solidity#207]
- Issue Type: `Performance`
- Extension Name: `solidity`
- Extension Version: `0.0.102`
- OS Version: `Windows_NT x64 10.0.18363`
- VSCode version: `1.52.1`

:warning: Make sure to **attach** this file from your *home*-directory:
:warning:`c:\Users\ALEXAN~1\AppData\Local\Temp\JuanBlanco.solidity-unresponsive.cpuprofile.txt`

Find more details here: https://github.com/microsoft/vscode/wiki/Explain-extension-causes-high-cpu-load

**Comment by juanfranblanco** [comment]
@SterlingKinder any more details?

**Comment by nelohenriq** [comment]
[JuanBlanco.solidity-unresponsive.cpuprofile.txt](https://github.com/juanfranblanco/vscode-solidity/files/8932755/JuanBlanco.solidity-unresponsive.cpuprofile.txt)

**Comment by nirban256** [comment]
Having the same problem, vs code is saying the extension took a long time to complete its previous task and is preventing other extensions to run in background whenever I am starting vs code, thereby increasing cpu load

**Comment by juanfranblanco** [comment]
What are your settings?

**Comment by nirban256** [comment]
I have all the default settings

**Comment by juanfranblanco** [comment]
What is your project structure, OS?

**Comment by nirban256** [comment]
I am using Windows 10, writing smart contracts with Hardhat for dapps and for frontend I am using ReactJs having the hardhat simple basic project structure

**Comment by juanfranblanco** [comment]
can you make an screenshot of your project structure? Your current version of vscode solidity? Is autocomplete working?

**Comment by nirban256** [comment]
My current version of vs code solidity  is `v0.0.139` and yes it is giving me suggestions for autocompleting my code most of the times but sometimes it is not showing those.

![Screenshot (1)](https://user-images.githubusercontent.com/74231771/175225151-e748751b-7b74-425d-b8f4-32803cf94563.png)

**Comment by juanfranblanco** [comment]
Hmmmm if you right click or press F5 to compile is it ok?

**Comment by nirban256** [comment]
yeah it is showing `Compilation completed successfully!`

**Comment by juanfranblanco** [comment]
Thank you, for all of this, I was just checking if there was something that might has caused this to be an issue, but nothing. 
Maybe this is due to trying to download the latest version of solidity when starting? Do you have any connectivity issues?

**Comment by juanfranblanco** [comment]
Also is this just happening on the start of vscode?

**Comment by nirban256** [comment]
No I am not having any connectivity issue, 
yes the error is showing only when I am starting my vs code

**Comment by juanfranblanco** [comment]
Is your solidity version set to latest? 
![image](https://user-images.githubusercontent.com/562371/175232383-9d1765b8-d19d-489c-99ec-4d10b6bc4156.png)

**Comment by nirban256** [comment]
yes it is set to latest

**Comment by juanfranblanco** [comment]
Ok, I am assuming this is due when trying to load and download the latest version on startup, this will happen when is starting as it will be initialised using the current settings. 

If you right click in a solidity smart contract and select this:
![image](https://user-images.githubusercontent.com/562371/175234362-13e68a8e-b5d1-4469-9bf9-f3cb195858a4.png)

And change it to 0.8.15 let me know if it makes any difference.

![image](https://user-images.githubusercontent.com/562371/175234503-c031b069-d7ea-4473-b0b0-652f25e61280.png)

**Comment by nirban256** [comment]
Ok

**Comment by nirban256** [comment]
I made the changes and the error is still showing

**Comment by juanfranblanco** [comment]
Thanks @nirban256, what is the spec of your PC? as this is not a normal issue.

**Comment by nirban256** [comment]
Processor	Intel(R) Core(TM) i5-2400 CPU @ 3.10GHz   3.10 GHz
Installed RAM	8.00 GB (7.70 GB usable)
System type	64-bit operating system, x64-based processor

**Comment by juanfranblanco** [comment]
Yes, that should not be an issue, if you don't have other processes hugging the cpu. Very strange, as you don't need to download the latest solidity and you are mainly loading the extension normally.

**Comment by nirban256** [comment]
No I am not running any other processes, hope that future updates will resolve the problem.
